src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.OSX.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal unsafe void SetCreationTime(string path, DateTimeOffset time)
+        {
+            Interop.Sys.TimeSpec timeSpec = default;
+
+            long seconds = time.ToUnixTimeSeconds();
+            long nanoseconds = UnixTimeSecondsToNanoseconds(time, seconds);
+
+            timeSpec.TvSec = seconds;
+            timeSpec.TvNsec = nanoseconds;
+
+            Interop.libc.AttrList attrList = default;
+            attrList.bitmapCount = Interop.libc.AttrList.ATTR_BIT_MAP_COUNT;
+            attrList.commonAttr = Interop.libc.AttrList.ATTR_CMN_CRTIME;
+
+            // Try to set the attribute on the file system entry using setattrlist,
+            // otherwise fall back to the method used on other unix platforms as the
+            // path may either be on an unsupported volume type (for setattrlist) or it
+            // could be another error (eg. no file) which the fallback implementation can throw.
+            bool succeeded = Interop.libc.setattrlist(path, &attrList, &timeSpec, sizeof(Interop.Sys.TimeSpec), new CULong(Interop.libc.FSOPT_NOFOLLOW)) == 0;
```

Comment:
Looking at this again, and I'm wondering: is it worth trying to fallback to the old unix implementation? I think we can just stick to using the OSX version, and then check the `errno` immediately after if `setattrlist` returned `false`.

The setattrlist manpage has a very complete list of errors to expect, so I don't think a fallback would add extra value: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/setattrlist.2.html

Reply:
Hi, I can do that if that's the best thing to do, I just saw the comment on that man page where it says `[ENOTSUP]          The volume does not support setattrlist().` and was concerned that some formats may not support this API (eg. FAT32 maybe or something like that), but may support others.

Reply:
Maybe we could keep the fallback for `ENOTSUP` only, and just throw it otherwise?